ckanext/ontario_theme/plugin.py

```diff
@@ -1175,7 +1175,10 @@ def before_search(self, search_params):
         # pages
         elif fq:
             fq_list = search_params.setdefault('fq_list', [default_open])
-            facet_al = default_open if "access_level" not in fq else ""
+            if (fq.count("access_level") == 1 and 'access_level:"open"' in fq) or "access_level" not in fq:
```

Comment:
@aleeexgreeen  If "Restricted" is selected along with "Open", and a search term is entered in the search box, the results will still pertain to "Restricted" and "Open" datasets. Is this correct?
```
/dataset/?q=forestry&access_level=open&access_level=restricted&sort=score+desc%2C+metadata_modified+desc
```

Reply:
It would work in this instance as well, but to provide more context, to make the dataset search pages open by default the facet query (fq/fq_list) needs to have `{!bool tag=orFqaccess_level should=\'access_level:"open"\'}`. So both when opening a dataset page and when the form is submitted and only access level open is selected, the facet query needs to have that item. When there are more than one access_level selected the facet query needs to have `{!bool tag=orFqaccess_level should=\'access_level:"<access_level>"\' should=\'access_level:"<access_level>"\'}`, so this is to make sure upon opening a dataset search page the default is set to open, if the form is submitted on only open is selected, the results pertain to only open and if any facet is added, the results pertain to all facets selected

Reply:
I simplified this section further